python/graphstorm/model/gnn_distill/student.py

```diff
@@ -0,0 +1,160 @@
+    limitations under the License.
+
+    GNN student model class
+"""
+import os
+import torch as th
+from torch import nn
+from transformers import (
+    AutoTokenizer,
+    AutoConfig,
+    AutoModel,
+    DistilBertConfig,
+    DistilBertModel,
+)
+
+SUPPORTED_MODEL = {
+"DistilBertModel": (DistilBertConfig, DistilBertModel),
+}
+
+PRETRAINED_MODEL = [
+"distilbert-base-uncased",
+"distilbert-base-cased",
+]
+
+class GSDistilledModel(nn.Module):
+    """ GraphStorm GNN distilled model class.
+    User specified Transformer-based model 
+    and a projection matrix will be initialized.
+
+    Parameters
+    ----------
+    lm_type : str
+        Model type for Transformer-based student model.
+    pre_trained_name : str
+        Name of pre-trained model.
+    checkpoint_path : str
+        Path of model checkpoint.
+    """
+    def __init__(self, lm_type=None, pre_trained_name=None, checkpoint_path=None):
+        super(GSDistilledModel, self).__init__()
+
+        # TODO: Try to unify between loading HF checkpoints and loading GS checkpoints.
+        if not checkpoint_path:
+            assert lm_type is not None, \
+                "HuggingFace Name of model architecture needs to be specified"
+            assert pre_trained_name is not None, \
+                "HuggingFace Name of pre-trained weights needs to be specified"
+            # TODO (HZ): need to test other HF models.
+            if lm_type not in SUPPORTED_MODEL:
+                raise ValueError(f'Model class {lm_type} is not supported.')
+            if pre_trained_name not in PRETRAINED_MODEL:
+                raise ValueError(f'Pre-trained model {pre_trained_name} is not supported.')
+
+            # initiate Transformer-based model
+            self.lm_config = SUPPORTED_MODEL[lm_type][0]()
+            self.lm = SUPPORTED_MODEL[lm_type][1](self.lm_config)
+
+            # load pre-trained tokenizers
+            self.tokenizer = AutoTokenizer.from_pretrained(pre_trained_name)
+
+            # load pre-trained parameters
+            self.lm = self.lm.from_pretrained(pre_trained_name)
+        else:
+            self.load_from_gs_checkpoint(checkpoint_path)
+
+        # TODO (HZ): support more distance-based loss
+        self.loss = nn.MSELoss()
+
+    def save_gs_checkpoint(self, checkpoint_path):
+        """ Save distilled checkpoint.
+        Parameters
+        ----------
+        checkpoint_path : str
+            Saved path for checkpoint.
+        """
+        proj_dir_loc = os.path.join(checkpoint_path, "proj")
+        os.makedirs(proj_dir_loc, exist_ok=True)
+        tokenizer_dir_loc = os.path.join(checkpoint_path, "tokenizer")
+        lm_dir_loc = os.path.join(checkpoint_path, "lm")
+        self.tokenizer.save_pretrained(tokenizer_dir_loc)
+        self.lm.save_pretrained(lm_dir_loc)
+        th.save(self.state_dict()["proj"], os.path.join(proj_dir_loc, "pytorch_model.bin"))
+
+    def load_from_gs_checkpoint(self, checkpoint_path):
```

Comment:
```suggestion
    def load_gs_checkpoint(self, checkpoint_path):
```

Reply:
updated.

Reply:
There is no place calling this function. Did you implement the functionality of resuming a distillation task?

Reply:
Not yet, but I tried load_gs_checkpoint on my end and it works well.

Reply:
Please open an issue for this and mark it 0.2.1